refinance/repository/base.py

```diff
@@ -1,59 +1,61 @@
 """Base repository with CRUD methods, suitable for simple objects"""
 
-from typing import Generic, Mapping, Type, TypeVar
+from abc import ABC, abstractmethod
+from typing import Any, Generic, Mapping, Type, TypeVar
 
 from fastapi import Depends
 from sqlalchemy.orm import Query, Session
 
 from refinance.db import get_db
 from refinance.errors.common import NotFoundError
 from refinance.models.base import BaseModel
-from refinance.schemas.base import PaginationSchema
+from refinance.schemas.base import BaseFilterSchema, PaginationSchema
 
-M = TypeVar("M", bound=BaseModel)  # model
-K = TypeVar("K", int, str)  # primary key
+_M = TypeVar("_M", bound=BaseModel)  # model
+_K = TypeVar("_K", bound=int | str)  # primary key
+_F = TypeVar("_F", bound=BaseFilterSchema)  # filters
+_Q = TypeVar("_Q", bound=Query[Any])  # query
```

Comment:
There is no need in Query typevar as it's always Query and nothing else. Also Any doesn't add typehint sense

Reply:
The thing is, Query is Generic. That is Query[query return-value]. So in order not to lose information about return-value in the function, I use generic to explicitly show that the return type of query does not change and that the time-checker works correctly after the function call.